lib/board.rb

```diff
@@ -0,0 +1,58 @@
+class Board
+
+  attr_accessor :cells
+
+  def initialize
+    @cells = Array.new(9, " ")
+  end
+
+  def display
+    puts " "
+    puts " #{@cells[0]} | #{@cells[1]} | #{@cells[2]} "
+    puts "-----------"
+    puts " #{@cells[3]} | #{@cells[4]} | #{@cells[5]} "
+    puts "-----------"
+    puts " #{@cells[6]} | #{@cells[7]} | #{@cells[8]} "
+    puts " "
+  end
+
+  def position(move)
+    @cells[index(move)]
+  end
+
+  def index(move)
+    move = move.to_i
+    move -= 1
+  end
+
+
+  def update(move, player)
+    @cells[index(move)] = player.token
+# in here I call player.token
+  end
+
+  def full?
+    @cells.all?{|i| i == "X" || i == "O"} ? true : false
+  end
+
+  def turn_count
+    turn_count = 0
```

Comment:
Use the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  cells.count { |cell| cell == "X" || cell == "O" }
end
```